src/EfuncMsg.chpl

```diff
@@ -101,6 +102,22 @@ module EfuncMsg
                         // code append second array's attrib
                         repMsg += "created " + st.attrib(rname2) + "+";
                     }
+                    when "popcount" {
+                        var a = st.addEntry(rname, e.size, int);
+                        a.a = popcount(e.a);
```

Comment:
@ronawho I tested this locally, but I wanted to double check: when promoting a scalar function like this over a block distributed array, will the compiler automatically use data parallelism?

Reply:
Correct, this compiler will transform this into something like:

```chpl
forall (_aa, _ea) in zip(a.a, e.a) {
  _aa = popcount(_ea);
}
```

https://chapel-lang.org/docs/users-guide/datapar/promotion.html and https://chapel-lang.org/docs/language/spec/data-parallelism.html#promotion have more descriptions of this behavior.

Reply:
Cool!